### DISCUSSION TITLE
CODEBLOCK formating!

### DISCUSSION THREAD

**Discussion by DavitTec** [zim-desktop-wiki/zim-desktop-wiki#2376] [Q&A]
Is there a easy way to control the **Codeblock** style and default formatting?

Say I wanted to resemble Terminal  Black background and white text for ALL or some Syntax code-block, like sh / shell.

Can this be setup from **_style.conf_**  ?

eg.
[CODEBLOCK]
default=sh
background=black
foreground=white
syntax=all
exclude=java, SQL
linenumbers=false
[java]
foreground=yellow
linenumbers=true

**Comment by dehesselle** [comment]
The code block is rendered by [GtkSourceView](https://wiki.gnome.org/Projects/GtkSourceView). You can modify or add [language definitions](https://wiki.gnome.org/Projects/GtkSourceView/LanguageDefinitions).

**Reply by introt** [reply]
The plugin's source hints at custom theme support, but the user-facing documentation is quite terse: https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/34b2d39a4aef83285e43840e12ffc47ec021e466/zim/plugins/sourceview.py#L51
